reconcile/utils/terraform_client.py

```diff
@@ -331,12 +331,27 @@ def log_plan_diff(
         for resource_change in resource_changes:
             resource_type = resource_change["type"]
             resource_name = resource_change["name"]
+            resource_previous_address = resource_change.get("previous_address")
             resource_change = resource_change["change"]
             actions = resource_change["actions"]
             for action in actions:
+                if resource_previous_address:
+                    # the resource is being moved/renamed in the TF state
+                    with self._log_lock:
+                        logging.info([
+                            "move/rename",
+                            name,
+                            resource_previous_address,
+                            resource_change["address"],
+                        ])
+
                 if action == "no-op":
                     logging.debug([action, name, resource_type, resource_name])
-                    continue
+                    if resource_previous_address:
+                        # apply resource renaming with no-op
+                        self.increment_apply_count()
```

Comment:
Do moved resources get applied each time or `resource_previous_address` is only going to be defined the first time the `moved` block appears and gets applied?

Reply:
I will double check this

Reply:
confirmed in a test with a `null_resource`:

initial `main.tf`, creating a `null_resource.null-1` resource
```
resource "null_resource" "null-1" {}
```

Rename resource to `null-2`
```
resource "null_resource" "null-2" {}

moved {
  from = null_resource.null-1
  to   = null_resource.null-2
}
```

second terraform run, showing `resource_previous_address` in the terraform plan output :
```
$ terraform plan -out plan.tf
null_resource.null-2: Refreshing state... [id=9133010116952026237]

Terraform will perform the following actions:

  # null_resource.null-1 has moved to null_resource.null-2
    resource "null_resource" "null-2" {
        id = "9133010116952026237"
    }

Plan: 0 to add, 0 to change, 0 to destroy.

$ terraform show -json plan.tfplan | jq '.resource_changes'
[
  {
    "address": "null_resource.null-2",
    "previous_address": "null_resource.null-1",
    "mode": "managed",
    "type": "null_resource",
    "name": "null-2",
    "provider_name": "registry.terraform.io/hashicorp/null",
    "change": {
      "actions": [
        "no-op"
      ],
      "before": {
        "id": "9133010116952026237",
        "triggers": null
      },
      "after": {
        "id": "9133010116952026237",
        "triggers": null
      },
      "after_unknown": {},
      "before_sensitive": {},
      "after_sensitive": {}
    }
  }
]

$ terraform apply plan.tf 

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

Once applied & renamed, second plan:
```
$ terraform plan -out plan.tfplan
null_resource.null-2: Refreshing state... [id=9133010116952026237]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

$ terraform show -json plan.tfplan | jq '.resource_changes'
[
  {
    "address": "null_resource.null-2",
    "mode": "managed",
    "type": "null_resource",
    "name": "null-2",
    "provider_name": "registry.terraform.io/hashicorp/null",
    "change": {
      "actions": [
        "no-op"
      ],
      "before": {
        "id": "9133010116952026237",
        "triggers": null
      },
      "after": {
        "id": "9133010116952026237",
        "triggers": null
      },
      "after_unknown": {},
      "before_sensitive": {},
      "after_sensitive": {}
    }
  }
]
```

so the `resource_previous_address` field is not there anymore since no move is needed

Reply:
awesome, thx for the detailed test